naptime/src/main/scala/org/coursera/naptime/access/StructuredAccessControl.scala

```diff
@@ -44,8 +45,231 @@ case class StructuredAccessControl[A](
           Authorizer.toResponse(authorizer.authorize(decorated), decorated)
         }
       }.getOrElse {
-        Left(NaptimeActionException(Status.UNAUTHORIZED, Some("auth.perms"), Some("Missing authentication")))
+        StructuredAccessControl.missingResponse
       }
     }
   }
 }
+
+object StructuredAccessControl extends StructuredAccessControlCombinators {
+  private[access] val missingResponse =
+    Left(NaptimeActionException(Status.UNAUTHORIZED, Some("auth.perms"), Some("Missing authentication")))
+}
+
+trait StructuredAccessControlCombinators {
+  /**
+   * A left-biased combinator that allows the request if either the left or the right access
+   * control allow it.
+   *
+   * Example usage:
+   * {{{
+   *   Nap.auth(
+   *     StructuredAccessControl.either(Auth.superuser, Auth.admin)).get(ctx => ???)
+   * }}}
+   *
+   * If there is only one parsing error, it is ignored. If there are two, the right one masks the
```

Comment:
Removed `parsing` as it is imprecise. The reason why is because when the left has an error, we fallback to whatever happens on the `Right`.